doc/build/sysbuild/index.rst

```diff
@@ -648,3 +693,285 @@ each image that is part of a project. Alternatively, there are
 which can be used to include CMake and Kconfig files for the overall sysbuild
 image itself, this is where e.g. a custom image for a particular board or SoC
 can be added.
+
+Grouping Zephyr applications in sysbuild
+****************************************
+
+When integrating multiple Zephyr applications into a single sysbuild project,
+you may find it useful to distinguish between different categories of images,
+such as bootloaders and non-bootloaders. Each image could be further divided
+according to which core it targets on whichever SoC on your physical board.
+As these groups emerge, often with some overlap, you may want all images in a
+particular group to share a set of properties, including a common configuration
+or a relationship to another group.
+
+You can use the ``ExternalZephyrProject_Group()`` function to arbitrarily define
+named groups of sysbuild domains. This function can both create new groups and
+progressively extend the definition of existing groups in-tree and out-of-tree.
+
+Defining sysbuild groups
+========================
+
+When adding a new Zephyr application to sysbuild, you can assign it to a group
+simply like this:
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Add(
+     APPLICATION my_sample
+     SOURCE_DIR <path-to>/my_sample
+     GROUP my_group
+   )
+
+Alternatively, you could use the ``ExternalZephyrProject_Group()`` function:
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Group(
+     other_group
+     INCLUDE my_sample
+   )
+
+In particular, if you are defining an out-of-tree group and you want to assign
+pre-existing application images to it, then this is the way to do that. At this
+point, ``my_sample`` belongs to both ``my_group`` and ``other_group``.
+
+As mentioned above, it is possible to extend the definition of existing groups
+in sysbuild. For instance, if you would like to add more images to ``my group``,
+then you can do it as follows:
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Group(
+     my_group
+     INCLUDE hci_rpmsg mcuboot
+   )
+
+Now, ``my_group`` contains ``my_sample``, ``hci_rpmsg``, and ``mcuboot``,
+provided that all of these images are present in the current multi-image build
+(non-existent images are quietly ignored). This works, because calling this
+function multiple times for the same group has a cumulative effect by default.
+
+On the other hand, if you want to actively prevent ``my_group`` from being
+extended like this, then you can define it as immutable:
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Group(
+     my_group
+     IMMUTABLE
+     INCLUDE my_sample
+   )
+
+An immutable group can only be defined once. In this case, since ``my_group``
+already exists, this would produce an error, so you could use a different name
+to ensure that your group has a consistent meaning in the build system.
```

Comment:
Should probably add a note with naming conventions for groups e.g. no spaces, semi-colons, commas, etc.

Reply:
Maybe. I suppose a note should be added somewhere for image names too.